### ISSUE TITLE
Images with opposing orientations and the Gallery Block

### ISSUE THREAD

**Issue by STHLMWebDesign** [WordPress/gutenberg#12258]
I have been testing Gutenberg on my photography blog and I really like it. I especially like the Gallery block which is incredibly useful on a website that deals with photography. 

However it has one major glaring omission. You can not place two pictures with opposing orientation next to one another. Gutenberg will by default crop into the pictures so they are the same size, if you turn the crop check box off then they two pictures will appear next to one another but one image will be taller than the other which does not look so nice. This is because when there is two, three, or more images Gutenberg wants to lay them out so they all have equal width. 

I really think Gutenberg should aim to do better than that. 

If I have two images I want to place next to one another, both of them are shot in 2*3 format however one is horizontal and one is vertical than I would like Gutenberg to be able to adjust the column width so they appear next to each other with the same height. Meaning the 2*3 frame in horizontal takes up most of the available space and the 2*3 shot in vertical takes up what remaining space is available so it matches the height of it’s companion picture. 

Hopefully I explained that well. 

Making posts in Gutenberg is a lot of fun and the Gallery block does a lot to allow you to make posts that are rich with photographs. But adding this feature would take it up to 11 and make it a killer block.

**Comment by STHLMWebDesign** [comment]
Here are some images to illustrate what I mean. 

This is the way Gutenberg adds images with different orientations and dimensions into a gallery block when you turn off cropping. Which isn't at all attractive. Plus I can say that as a photographer cropping into the images to make them all the same size is also not an attractive option.   

<img width="1524" alt="screenshot" src="https://user-images.githubusercontent.com/1607670/49221237-d792ae80-f3d8-11e8-9e77-d514abfb6792.png">

This is what I would like to be able to do in Gutenberg. The original dimensions of the images are maintained and they all are shrunk down so they fit into a row together: 

![example](https://user-images.githubusercontent.com/1607670/49221265-eaa57e80-f3d8-11e8-88a4-c5c4ef9ffd04.jpg)

**Comment by youknowriad** [comment]
Thanks for your suggestion, I wonder if  this should be something for an advanced third-party Gallery plugin. Thoughts on this @mapk @jasmussen

**Comment by jasmussen** [comment]
I believe, correct me if I'm wrong, that this is exactly what the Jetpack Tiled Gallery does. Alternately you can try a little custom CSS, I think what's on http://moc.co/2018/02/masonry-galleries-gutenberg/ might work for this.

I do agree that the un-cropped version of images in the stock gallery is a little uninspired, so I wouldn't be opposed to improving that aspect of the gallery. In your example you'll see that all 3 columns are exactly 33% wide, causing items inside to scale suboptimally. 

In any case it would be somewhat difficult to refactor the gallery block to do this, which means were we to do it it would take some time. For that reason you might want to look into some plugin blocks in the mean time.

**Comment by mapk** [comment]
I agree that the Gallery block without the "crop images" toggle turned on, it looks awkward. In the future, an option to tile the gallery might be a cool style variation, but it would be down the road.

I'll move this to the `future` milestone.

**Comment by pfroud** [comment]
+1 for this Gutenberg enhancement. I cooked up some CSS to make the Gallery block work better with images of opposing orientations / aspect ratios:

```
/* Manually set the gallery heights depending on how many columns */
figure.wp-block-gallery.columns-2 figure {
	height: 300px;
}
figure.wp-block-gallery.columns-3 figure {
	height: 250px;
}
figure.wp-block-gallery.columns-4 figure {
	height: 200px;
}
figure.wp-block-gallery.columns-5 figure {
	height: 150px;
}
figure.wp-block-gallery.columns-6 figure {
	height: 150px;
}


figure.wp-block-gallery ul.blocks-gallery-grid {
	/* In some cases, makes the images evenly distributed horizontally */
	width: 100%;
}

figure.wp-block-gallery figure {
	/* Vertically center images in container */
	align-items: center !important;
}

figure.wp-block-gallery img {
	/* Make images stay inside container */
	max-width: 100%;
	max-height: 100%;
	
	/* Horizontally center images in container */
	margin-left: auto;
	margin-right: auto;
}
```

Joen, the CSS you linked to didn't work for me, but it turns out the Jetpack Tiled Gallery does this very well. However a big plugin seems like a lot just to use the gallery.

Screenshot of the Jetpack Tiled Gallery:
![Jetpack tiled gallery](https://user-images.githubusercontent.com/9056906/79628529-afebd580-80f6-11ea-8581-f0d9380064ca.png)


Screenshot of the Gutenberg Gallery block without custom CSS:
![wordpress gallery block without custom CSS](https://user-images.githubusercontent.com/9056906/79628549-c7c35980-80f6-11ea-812e-f5ab5ddd1eca.png)


Screenshot of the Gutenberg Gallery block with my custom CSS:
![Wordpress gallery block with custom CSS](https://user-images.githubusercontent.com/9056906/79628560-d578df00-80f6-11ea-8a71-86b1be676af1.png)

&nbsp;

Also for comparison, here's a screenshot of the Grid Gallery available in Squarespace:
![Squarespace grid gallery](https://user-images.githubusercontent.com/9056906/79628700-38b74100-80f8-11ea-8e22-96f3662f2097.png)

I tried to copy the Squarespace gallery when doing my CSS.

The controls let you choose how many columns, like the Gutenberg gallery. However you also choose the aspect ratio for the thumbnails which is weird.
![Squarespace grid gallery controls](https://user-images.githubusercontent.com/9056906/79628726-761bce80-80f8-11ea-806e-ac3b4eea36c6.png)

![Squarespace grid gallery aspect ratio menu](https://user-images.githubusercontent.com/9056906/79628736-8cc22580-80f8-11ea-9232-15b54d80fecb.png)

**Comment by mtias** [comment]
The new aspect ratio controls should allow this to work. Please reopen if there are still cases to address.
